### PR TITLE
nn: init at 2.0.8-unstable-2024-04-08

### PR DESCRIPTION
**Pull request by mk3z**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/nn/nn/package.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, lib, fetchFromGitHub }:
+stdenv.mkDerivation rec {
```
Comment (JohnRTitor) [inline pkgs/by-name/nn/nn/package.nix:2]: using `rec`ursion is discouraged, use `finalAttrs` instead`

File: pkgs/by-name/nn/nn/package.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  name = "nn";
+  version = "1.85.0";
```
Comment (JohnRTitor) [inline pkgs/by-name/nn/nn/package.nix:4]: 1.85.0 is the "last official release", released in 2015, 9 years ago.

Looks like there is a [recent release](https://github.com/sakov/nn-c/commit/8e9ce3f39b8d0ae034215be19c8688bdb285aa8a), though, it's not tagged, can we use that?

Comment (JohnRTitor) [reply pkgs/by-name/nn/nn/package.nix:4]: Though for this one, I would recommend using the latest commit all the time, if there is not any issues. We can use `unstableGitUpdater` - https://github.com/NixOS/nixpkgs/blob/03c07e5a7a08a24d24bd0d0d79d0b98d3e657c18/pkgs/common-updater/unstable-updater.nix to easily update it ourselves or bots bump this

**Review by JohnRTitor — COMMENTED** [review]
In additon, format it using `nixfmt-rfc-style

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 325507` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nn</li>
  </ul>
</details>

**Review by JohnRTitor — APPROVED** [review]
LGTM

**Comment by mk3z** [comment]
I changed the description to fit the guidelines in https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

**Review by Sigmanificient — APPROVED** [review]

**Pull request merged**